devices.go

```diff
@@ -172,10 +183,48 @@ func (d *deviceCache) SetDeviceByIdOpState(id string, os models.OperatingState)
 
 // Update updates the device in the cache and ensures that the
 // copy in Core Metadata is also updated.
-func (d *deviceCache) Update(id string) error {
+func (d *deviceCache) Update(dev *models.Device) error {
+	addressable, err := updateAddressable(&dev.Addressable)
+	if err != nil {
+		return err
+	}
+	dev.Addressable = *addressable
+
+	device, err := svc.dc.DeviceForName(dev.Name)
+	if err != nil {
+		return err
+	}
+
+	dev.Id = device.Id
+	err = svc.dc.Update(*dev)
+	if err != nil {
+		return err
+	}
+
+	d.devices[dev.Name] = dev
```

Comment:
We will allow modifying device name? If we will allow it, this code should be modified (remove old one, add new one). Otherwise, maybe should be fine to check if it's trying to modify the name, and return an error in this case.
Another option it's just call `RemoveById()` and then `Add()`

Reply:
I make some change. First check names map and then put new value to devices map and names map. Please help review, thanks.